### PR TITLE
First pass at native mark-for-deployment slack notifications

### PR DESCRIPTION
**Pull request by solarkennedy**

### CODE REVIEW

**Comment by solarkennedy** [comment]
@EvanKrall can you review this please.

File: paasta_tools/cli/cmds/mark_for_deployment.py [hunk 1]
```diff
@@ -202,6 +206,103 @@ def report_waiting_aborted(service, deploy_group):
...
+
+    def deploy_group_is_set_to_notify(self):
+        for step in self.deploy_info.get('pipeline', []):
+            if step.get('step', '') == self.deploy_group:
+                return step.get('slack_notify', False)
+        return False
+
+    def post(self, **kwargs):
+        if self.deploy_group_is_set_to_notify():
+            self.sc.post(**kwargs)
```
Comment (EvanKrall) [inline paasta_tools/cli/cmds/mark_for_deployment.py]: doesn't look like you're actually calling this method anywhere; you're calling `self.sc.post` in the methods that follow.

File: paasta_tools/cli/cmds/mark_for_deployment.py [hunk 2]
```diff
@@ -202,6 +206,103 @@ def report_waiting_aborted(service, deploy_group):
...
+        self.sc.post(channels=self.channels, message=message)
+        if self.old_commit is not None and self.commit != self.old_commit:
+            mes = (
+                "If you need to roll back, run:\n"
+                f"```paasta rollback --service {self.service} --deploy-group {self.deploy_group} --commit {self.commit}"
+            )
+            self.sc.post(channels=self.channels, message=mes)
+
+
+def validate_commit(commit, git_url):
```
Comment (EvanKrall) [inline paasta_tools/cli/cmds/mark_for_deployment.py]: I think this code is also copy-pasted into wait_for_deployment.py - you could make that call this function.

File: paasta_tools/cli/cmds/mark_for_deployment.py (hunk 1 above)
Comment (solarkennedy) [reply paasta_tools/cli/cmds/mark_for_deployment.py]: whoops. Lack of testing contributed to this bug.

I've added a test and corrected it.

File: paasta_tools/cli/cmds/mark_for_deployment.py (hunk 2 above)
Comment (solarkennedy) [reply paasta_tools/cli/cmds/mark_for_deployment.py]: Hmm yea, will do

**Pull request merged**